polling-app-server/src/main/java/com/example/polls/controller/AuthController.java

```diff
@@ -51,9 +59,19 @@
     @Autowired
     JwtTokenProvider tokenProvider;
 
+    @Autowired
+    private MailService mailService;
+
+    private static final Logger logger = LoggerFactory.getLogger(AuthController.class);
+
     @PostMapping("/signin")
     public ResponseEntity<?> authenticateUser(@Valid @RequestBody LoginRequest loginRequest) {
 
+        Optional<User> user = userRepository.findByUsername(loginRequest.getUsernameOrEmail());
```

Comment:
guy,you know what is Optional right? why not use user.isPresent()?this will cause null pointer exception!

Reply:
Point taken. Will update

Reply:
But you could have said that in a polite way.